frontend/src/components/utils/MultiUserSelector.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useContext, useEffect, useState } from "react";
+import { api, useBetterQuery } from "../../services/apiService";
+import Select from "react-select";
+import { ErrorMessage } from "./ErrorPage";
+import { Loading } from "./Loading";
+import { Col, Row } from "react-bootstrap";
+import { User } from "../../models/user";
+import { UserAvatarCard } from "./avatar/UserAvatarCard";
+import { Size } from "../../utils/size";
+import { UserContext } from "../../services/authService";
+import { DebounceInput } from "react-debounce-input";
+
+export const MultiUserSelector = ({ onChange }) => {
+    const loggedUser = useContext(UserContext);
+
+    const [searchValue, setSearchValue] = useState("");
+    const [promotion, setPromotion] = useState<any>(null);
+
+    const [userStatus, setUserStatus] = useState<
+        "loading" | "error" | "success"
+    >();
+    const [userError, setUserError] = useState<any>();
+
+    const [users, setUsers] = useState<{
+        [key: string]: { user: User; selected: boolean };
+    }>({});
```

Comment:
What this state contains is a bit shady. It's the selected users AND the users belonging to the selected promotion.
It's quite reasonable but please explain that.